### PR TITLE
Conversations

### PR DESCRIPTION
**Pull request by fpariani**
When a target is created, the compatibles ones are searched and then a conversation between the pair of users is created. Also it's possible to get all of user's conversations.

### CODE REVIEW

File: app/controllers/api/v1/targets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    class TargetsController < Api::V1::ApiController
+      def index
+        @targets = current_user.targets
+      end
+
+      def create
+        @target = current_user.targets.create!(target_params)
+        @matchers = []
+        Geocoder.configure(units: :km)
+        Target.where('user_id != ?', current_user.id).each do |t|
```
Comment (matiasmansilla1989) [inline app/controllers/api/v1/targets_controller.rb]: `Target.where.not(user_id: current_user.id)`

File: app/controllers/api/v1/targets_controller.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+module Api
+  module V1
+    class TargetsController < Api::V1::ApiController
+      def index
+        @targets = current_user.targets
+      end
+
+      def create
+        @target = current_user.targets.create!(target_params)
```
Comment (matiasmansilla1989) [inline app/controllers/api/v1/targets_controller.rb]: We should move most of this logic into a new service.

File: app/controllers/api/v1/targets_controller.rb [hunk 3]
```diff
@@ -7,6 +7,12 @@ def index
 
       def create
         @target = current_user.targets.create!(target_params)
+        @matchers = []
+        Geocoder.configure(units: :km)
```
Comment (jiv) [inline app/controllers/api/v1/targets_controller.rb]: I think you can move this configuration to an initializer, check here: https://github.com/alexreisner/geocoder

File: app/controllers/api/v1/targets_controller.rb [hunk 4]
```diff
@@ -7,6 +7,12 @@ def index
 
       def create
         @target = current_user.targets.create!(target_params)
+        @matchers = []
```
Comment (jiv) [inline app/controllers/api/v1/targets_controller.rb]: I would assign this variable as the return of the `compare_distances` function. That way it's more explicit where this variable is loaded

File: spec/requests/api/v1/targets/create_spec.rb [hunk 5]
```diff
@@ -43,4 +43,55 @@
...
+                                        area_length: 600, topic_id: other_user_target.topic.id)
+      }
+    end
+
+    it 'returns success after create a target for user' do
+      post api_v1_targets_path,
+           params: create_target_params,
+           headers: user_auth_headers,
+           as: :json
+      expect(response).to have_http_status(200)
```
Comment (jiv) [inline spec/requests/api/v1/targets/create_spec.rb]: I thought you were using :ok instead of 200

File: spec/requests/api/v1/users/logout_spec.rb [hunk 6]
```diff
@@ -1,8 +1,8 @@
 require 'rails_helper'
+require 'support/request_spec_helper'
```
Comment (jiv) [inline spec/requests/api/v1/users/logout_spec.rb]: Not necessary, you added this require in spec_helper

**Pull request merged**